### PR TITLE
Add Dormant Microsoft Defender

### PR DESCRIPTION
**Pull request by espanafly**
Purpose described in "About Dormant Microsoft Defender" section of "README.md"

### CODE REVIEW

File: Src/Offline-Resources.psm1 [hunk 1]
```diff
@@ -54,6 +54,7 @@ $OptimizeOffline.DevMode = (Resolve-FullPath -Path $OptimizeOffline.Packages -Ch
 $OptimizeOffline.WindowsStore = (Resolve-FullPath -Path $OptimizeOffline.Packages -Child WindowsStore)
 $OptimizeOffline.MicrosoftEdge = (Resolve-FullPath -Path $OptimizeOffline.Packages -Child MicrosoftEdge)
 $OptimizeOffline.Win32Calc = (Resolve-FullPath -Path $OptimizeOffline.Packages -Child Win32Calc)
+$OptimizeOffline.DormantDefender = (Resolve-FullPath -Path $OptimizeOffline.Packages -Child DormantDefender)
```
Comment (gdeliana) [inline Src/Offline-Resources.psm1]: Can you please explain me what shall the Packages/DormantDefender folder serve for?

File: en-US/Optimize-Offline-help.xml [hunk 2]
```diff
@@ -453,7 +476,7 @@
       </command:example>
       <command:example>
         <maml:title>-------------------------- EXAMPLE 2 --------------------------</maml:title>
-        <dev:code>Optimize-Offline -SourcePath "D:\Images\Windows 10 1903\18362.1.190318-1202.19H1_RELEASE_CLIENTMULTI_X64FRE_EN-US.iso" -WindowsApps "Select" -SystemApps -Capabilities -Packages -Features -Win32Calc -Dedup -DaRT "Setup" -Registry -Additional @{ Setup = $true; RegistryTemplates = $true; Drivers = $true } -ISO "No-Prompt"</dev:code>
+        <dev:code>Optimize-Offline -SourcePath "D:\Images\Windows 10 1903\18362.1.190318-1202.19H1_RELEASE_CLIENTMULTI_X64FRE_EN-US.iso" -WindowsApps "Select" -SystemApps -Capabilities -Packages -Features --Win32Calc -DormantDefender -Dedup -DaRT "Setup" -Registry -Additional @{ Setup = $true; RegistryTemplates = $true; Drivers = $true } -ISO "No-Prompt"</dev:code>
```
Comment (gdeliana) [inline en-US/Optimize-Offline-help.xml]: Please correct the --Win32Calc typo

**Review by gdeliana — CHANGES_REQUESTED** [review]
Overall i like the idea. I just had one small question. What happens if the user decides to remove the SecHealthUIPackage and at the same time choses DormantDefender option to ON? Shall we maybe make also a condition in the app/sysapp removal code for not letting the package getting removed even when the user has selected it for removal when he enabled the DormantDefender option?

**Comment by espanafly** [comment]
thanks for the feedback!
I did consider that someone could remove the SecHealthUI package and at the same time choose the DormantDefender option. 
For my level of coding skills, I could see putting a pause in the script and add a warning that "You've chosen both. What do you really want to do?" 
I decided to just leave it and let the user choose both. At a code-level there isn't a conflict since SecHealthUI package removal would be the dominant outcome. 
I can see that the user might be surprised to discover that Defender is not just dormant, but actually unrecoverable (gone).
My first reaction to that is: "Too bad! RTFM!" But maybe that is harsh?

In defense of the "Too bad! RTFM!" solution. The user isn't going to get much benefit choosing DormantDefender TRUE without having read the ReadMe section and downloaded the online re-activation and Toggle scripts. In my mind, the README.md section makes it clear that choosing DormantDefender and also removing the SecHealthUI package is pointless. 

I still prefer your idea of integrating the less extreme registry and service changes performed by ToggleDefender and then including ToggleDefender in the O-O package. Your integration idea also has to figure out how to prevent a user from removing the SecHealthUI package.

'Src/Offline-Resources.psm1'
Please ignore that file change. There is no need for a folder.
'en-US/Optimize-Offline-help.xml'
I'll correct typo to and resubmit

**Comment by gdeliana** [comment]
> thanks for the feedback! I did consider that someone could remove the SecHealthUI package and at the same time choose the DormantDefender option. For my level of coding skills, I could see putting a pause in the script and add a warning that "You've chosen both. What do you really want to do?" I decided to just leave it and let the user choose both. At a code-level there isn't a conflict since SecHealthUI package removal would be the dominant outcome. I can see that the user might be surprised to discover that Defender is not just dormant, but actually unrecoverable (gone). My first reaction to that is: "Too bad! RTFM!" But maybe that is harsh?
> 
> In defense of the "Too bad! RTFM!" solution. The user isn't going to get much benefit choosing DormantDefender TRUE without having read the ReadMe section and downloaded the online re-activation and Toggle scripts. In my mind, the README.md section makes it clear that choosing DormantDefender and also removing the SecHealthUI package is pointless.
> 
> I still prefer your idea of integrating the less extreme registry and service changes performed by ToggleDefender and then including ToggleDefender in the O-O package. Your integration idea also has to figure out how to prevent a user from removing the SecHealthUI package.
> 
> 'Src/Offline-Resources.psm1' Please ignore that file change. There is no need for a folder. 'en-US/Optimize-Offline-help.xml' I'll correct typo to and resubmit

Ok perfect let's leave it to the user then, no need to put extra logic. Just remove the extra path in the resources. I will integrate your PR and then add the Toggle- Defender as a called subscript of set-additional.

**Comment by espanafly** [comment]
> Ok perfect let's leave it to the user then, no need to put extra logic. Just remove the extra path in the resources. I will integrate your PR and then add the Toggle- Defender as a called subscript of set-additional.

Thanks mate!
I cleaned up the formatting and added a "Note" paragraph to the "Dormant Microsoft Defender" section of the README.md. Hopefully my additional words, makes it clear what happens if both control methods are applied.
I'll remove the extra path in the resources and commit that change.
Thanks for all the work you do to keep this project alive!

**Comment by gdeliana** [comment]
Mate i think you have shaved all the bugs ;) 
Really a lot of thanks for your hard work!!!

**Comment by espanafly** [comment]
No worries, mate!
This was my first attempt at making a parameter. I picked something easy and hopefully useful. Thank you for your help and advice and for all the effort you make to continually advance this project!

**Pull request merged**